zokrates_core/src/flatten/mod.rs

```diff
@@ -724,6 +724,33 @@ impl<'ast, T: Field> Flattener<'ast, T> {
 
                 FlatExpression::Identifier(sub_bits_be[bitwidth - 1])
             }
+            BooleanExpression::BoolEq(box lhs, box rhs) => {
+                // Wanted: Not((x-y)^2)
```

Comment:
Can we put the full proof here? something like if lhs and rhs are booleans, they flatten to 1 or 0. Wanted truth table: ... . Truth table of Not((flat(x) - flat(y))**2): ... .